config/deploy/rbac.yaml

```diff
@@ -46,7 +46,26 @@ apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
   name: reference-addon
-rules: []
+rules:
+- apiGroups:
+  - "addons.managed.openshift.io"
+  resources:
+  - addoninstances/status
+  verbs:
+  - get
+  - list
+  - watch
+  - update
+  - patch
+- apiGroups:
+  - "addons.managed.openshift.io"
+  resources:
+  - addons
+  resourceNames:
+  - reference-addon ## name of the addon
+  verbs:
+  - get
+  - list
```

Comment:
I don't think an Addon should have to interact with it's own Addon object and it should even less concern itself with other Addons on the cluster.

Reply:
so, the Addon would be making use of AddonInstance client SDK which, behind the scenes, would need to perform a GET to the Addon and parse the target namespace corresponding to it. Rest, the addon (and its developers) need not explicitly interact with its addon object.

Reply:
Because the client is running within the target namespace (As we install the operator into it).
I would get the namespace from the downward API, just to make everything way less complicated:

E.g.:
```yaml
env:
  - name: MY_POD_NAMESPACE
  valueFrom:
    fieldRef:
    fieldPath: metadata.namespace
```

We can also consider using the new config feature to inject the targetNamespace via the Subscription, but this is also making it more complicated again.

Reply:
The env-var population via downwards API's solution sounds pretty neat! Doesn't seem to require a massive overhead for the tenants as well, all they gotta do is just throw 
```
- name: ADDON_TARGET_NAMESPACE
  valueFrom:
    fieldRef:
      fieldPath: metadata.namespace
```
under their addon's deployment.spec.template.spec

Reply:
We still need to drop RBAC for addons here :)

Reply:
Whoops, forgot to remove that, will do :)